modal_proto/api.proto

```diff
@@ -2875,6 +2875,7 @@ message VolumeMount {
   string volume_id = 1;
   string mount_path = 2;
   bool allow_background_commits = 3;
+  bool read_only = 4;
```

Comment:
Just to overthink things, is there any more general concept of "mode" or "permissions" that could end up being more useful than just having a boolean?

Reply:
I don't think there are any related non-orthogonal concerns. It's either read-only or read-write (there's no write-only mode).

There are other mount options you could theoretically want to set, like `NoExec`. But as long as the mount options aren't duplicated in a ton of different proto messages just adding another `bool` for that seems fine to me.

I think we can cross that bridge when we get there.